doc/util.md

```diff
@@ -241,3 +242,34 @@ Operators:
  - `&`: Combines two `Vector2`s to form a Rect, the origin should be on the left and the size on the
   right.
  - `%`: Modulo/Remainder of x and y separately of two `Vector2`s.
+
+### Matrix4
+
+This class comes from the `vector_math` package. We have created a few extension methods on top
+of what is already offered by `vector_math`.
+
+Methods:
+  - `transform2`: Create a new `Vector2` by transforming the given `Vector2` using the `Matrix4`.
+  - `transformed2`: Transform the input `Vector2` into the output `Vector2`.
+
+Getters:
+  - `m11`: The first row and first column.
+  - `m12`: The first row and second column.
+  - `m13`: The first row and third column.
+  - `m14`: The first row and fourth column.
+  - `m21`: The second row and first column.
+  - `m22`: The second row and second column.
+  - `m23`: The second row and third column.
+  - `m24`: The second row and fourth column.
+  - `m31`: The third row and first column.
+  - `m32`: The third row and second column.
+  - `m33`: The third row and third column.
+  - `m34`: The third row and fourth column.
+  - `m41`: The fourth row and first column.
+  - `m42`: The fourth row and second column.
+  - `m43`: The fourth row and third column.
+  - `m44`: The fourth row and fourth column.
```

Comment:
can't we just say mXY returns row and col y

Reply:
When writing these docs I was wondering... why are we adding all of this when we can just link to our generated API's? Especially for these extensions classes there is no extra value (like explaining usecases) to define each method and getter explicitly as the API docs will already do that for us.